Data missing in transport timeseries when running the default config

### What happened?

While the minimal config runs well for the implementation of the transport sector, the default config fails because:

- the annual timeseries profile data from RAMP is missing CYP. This is an easy fix by adding `CYP: ['ROU']` in the fill-missing-values dictionary in the config file
- some annual demand data for historic-electrification is missing for CHE. The file `build/data/transport/annual-road-transport-historic-electrification.csv` does not come with CHE. See log below. 

### Version

1.2.0.dev

### Relevant log output

```
(euro-calliope) adrienmellot@Adriens-MBP euro-calliope % snakemake --profile profiles/conda --configfile="config/default.yaml" "build/data/transport/timeseries/timeseries-passenger-cars-historic-electrification.csv"

Config file config/default.yaml is extended by additional config specified via the command line.
Building DAG of jobs...
Using shell: /bin/bash
Provided cores: 2
Rules claiming more threads will be scaled down.
Job stats:
job                                                          count    min threads    max threads
---------------------------------------------------------  -------  -------------  -------------
create_road_transport_timeseries_historic_electrification        1              1              1
total                                                            1              1              1

Select jobs to execute...

[Thu Mar 21 15:50:44 2024]
Job 0: Create timeseries for historic electrified road transport demand
Reason: Missing output files: build/data/transport/timeseries/timeseries-passenger-cars-historic-electrification.csv

Activating conda environment: ../envs/snakemake/euro-calliope/532acc2ee043cde2d8eb0d26b6750c89_
Activating conda environment: ../envs/snakemake/euro-calliope/532acc2ee043cde2d8eb0d26b6750c89_
/Users/adrienmellot/Documents/GitHub/envs/snakemake/euro-calliope/532acc2ee043cde2d8eb0d26b6750c89_/lib/python3.9/site-packages/numpy/lib/arraysetops.py:583: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  mask |= (ar1 == a)

Traceback (most recent call last):
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/.snakemake/scripts/tmp2ebh8nme.road_transport_timeseries.py", line 84, in <module>
    create_road_transport_demand_timeseries(
  File "/Users/adrienmellot/Documents/GitHub/euro-calliope/.snakemake/scripts/tmp2ebh8nme.road_transport_timeseries.py", line 70, in create_road_transport_demand_timeseries
    assert not df_timeseries.isna().any(
AssertionError: There are NaN values in the timeseries dataframe

[Thu Mar 21 15:50:54 2024]
Error in rule create_road_transport_timeseries_historic_electrification:
    jobid: 0
    input: build/data/transport/annual-road-transport-historic-electrification.csv, data/automatic/ramp-ev-consumption-profiles.csv.gz
    output: build/data/transport/timeseries/timeseries-passenger-cars-historic-electrification.csv
    conda-env: /Users/adrienmellot/Documents/GitHub/envs/snakemake/euro-calliope/532acc2ee043cde2d8eb0d26b6750c89_

Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-03-21T155043.561813.snakemake.log
```

--- a/config/default.yaml
+++ b/config/default.yaml
@@ -38,6 +38,7 @@ data-pre-processing:
             BIH: [HRV, HUN]
             MNE: [HRV]
             SRB: [HUN]
+            CYP: [ROU]
 root-directory: .
 cluster-sync:
     url: euler.ethz.ch

--- a/scripts/data/annual_energy_balance.py
+++ b/scripts/data/annual_energy_balance.py
@@ -194,21 +194,16 @@ def get_ch_waste_consumption(path_to_excel):
 
 def get_ch_transport_energy_balance(path_to_excel):
     carriers = {
-        "Elektrizität": "E7000",
-        "Gas übriger Vekehr": "G3000",
-        "Übrige erneuerbare Energien": "R5220B",
-        "davon Benzin": "O4652XR5210B",
-        "davon Diesel": "O4671XR5220B",
-        "davon Flugtreibstoffe": "O4000XBIO",
+        "Gas übriger Vekehr": ("G3000", "FC_TRA_ROAD_E"),
+        "Übrige erneuerbare Energien": ("R5220B", "FC_TRA_ROAD_E"),
+        "davon Benzin": ("O4652XR5210B", "FC_TRA_ROAD_E"),
+        "davon Diesel": ("O4671XR5220B", "FC_TRA_ROAD_E"),
+        "davon Flugtreibstoffe": ("O4000XBIO", "INTAVI"),
+        "davon Bahnen": ("E7000", "FC_TRA_RAIL_E"),
+        "davon Strasse": ("E7000", "FC_TRA_ROAD_E"),
     }
-    categories = {
-        "O4652XR5210B": "FC_TRA_ROAD_E",
-        "O4671XR5220B": "FC_TRA_ROAD_E",
-        "R5220B": "FC_TRA_ROAD_E",
-        "G3000": "FC_TRA_ROAD_E",
-        "E7000": "FC_TRA_RAIL_E",
-        "O4000XBIO": "INTAVI",
-    }
+    # ASSUME "davon Non-Road" is not included in electrified transport
+
     df = pd.read_excel(
         path_to_excel,
         skiprows=6,
@@ -219,6 +214,7 @@ def get_ch_transport_energy_balance(path_to_excel):
         na_values=["-"],
         header=[0, 1, 2, 3, 4],
     ).xs("TJ", level=-1, axis=1)
+
     # Footnote labels lead to some strings randomly ending in numbers; we remove them here
     remove_digits = str.maketrans("", "", digits)
     # carrier names span across two column levels, which we merge with fillna
@@ -233,14 +229,15 @@ def get_ch_transport_energy_balance(path_to_excel):
 
     df.columns = carrier_name_func(0).fillna(carrier_name_func(1)).values
 
-    return (
-        df.groupby(axis=1, level=0)
+    df = (
+        df
+        .groupby(axis=1, level=0)
         .sum()
         .rename_axis(index="year", columns="carrier_code")
-        .T.assign(cat_code=lambda df: df.index.map(categories))
-        .set_index("cat_code", append=True)
-        .stack()
+        .T
     )
+    df.index = pd.MultiIndex.from_tuples(df.index, names=('carrier_code', 'cat_code'))
+    return df.stack()
 
 
 def get_ch_industry_energy_balance(path_to_excel):

--- a/scripts/data/annual_energy_balance.py
+++ b/scripts/data/annual_energy_balance.py
@@ -230,13 +230,12 @@ def get_ch_transport_energy_balance(path_to_excel):
     df.columns = carrier_name_func(0).fillna(carrier_name_func(1)).values
 
     df = (
-        df
-        .groupby(axis=1, level=0)
+        df.groupby(axis=1, level=0)
         .sum()
         .rename_axis(index="year", columns="carrier_code")
         .T
     )
-    df.index = pd.MultiIndex.from_tuples(df.index, names=('carrier_code', 'cat_code'))
+    df.index = pd.MultiIndex.from_tuples(df.index, names=("carrier_code", "cat_code"))
     return df.stack()
 
 

--- a/scripts/transport/annual_transport_demand.py
+++ b/scripts/transport/annual_transport_demand.py
@@ -65,7 +65,6 @@ def get_all_distance_efficiency(
         .interpolate(axis=1, limit_direction="both")
         .stack()
     )
-
     # contribution of each transport mode to carrier consumption from JRC_IDEES
     # 2016-2018 from 2015 data; non-JRC countries, based on neighbour data
     carrier_contribution = fill_missing_countries_and_years(
@@ -74,7 +73,6 @@ def get_all_distance_efficiency(
         ),
         fill_missing_values,
     )
-
     # Energy consumption per transport mode by mapping transport mode
     # carrier contributions to total carrier consumption
     transport_energy_per_mode = carrier_contribution.mul(

--- a/scripts/transport/road_transport_timeseries.py
+++ b/scripts/transport/road_transport_timeseries.py
@@ -61,7 +61,6 @@ def create_road_transport_demand_timeseries(
         .tz_localize(None)
         .rename_axis("utc-timestamp")
     )
-
     assert not df_timeseries.isna().any(
         axis=None
     ), "There are NaN values in the timeseries dataframe"

